src/org/zaproxy/zap/extension/codedx/CodeDxAPI.java

```diff
@@ -0,0 +1,150 @@
+package org.zaproxy.zap.extension.codedx;
+
+import net.sf.json.JSONObject;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.log4j.Logger;
+import org.zaproxy.zap.extension.api.*;
+import org.zaproxy.zap.extension.api.ApiException.Type;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+public class CodeDxAPI extends ApiImplementor {
+
+	private static final Logger LOGGER = Logger.getLogger(CodeDxExtension.class);
+
+	private static final String PREFIX = "codedx";
+
+	private static final String VIEW_GENERATE = "generateReport";
+	private static final String ACTION_UPLOAD = "uploadReport";
+	private static final String ACTION_GEN_UPLOAD = "generateAndUpload";
+
+	private static final String ACTION_PARAM_FILE_PATH = "filePath";
+	private static final String ACTION_PARAM_SERVER_URL = "serverUrl";
+	private static final String ACTION_PARAM_API_KEY = "apiKey";
```

Comment:
I'd suggest being more specific (`codeDxApiKey`?), it might be misleading as is, looks a lot like the `apikey` parameter (for the ZAP API).